From 2494f0f0d73ab9be1ddea5668d4fcf084cba3355 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:26 -0400
Subject: [ethernet] cxgb4: Print adapter VPD Part Number instead of Engineering Change field

Message-id: <1410475447-94392-8-git-send-email-svemuri@redhat.com>
Patchwork-id: 93844
O-Subject: [RHEL7.1 PATCH BZ 1124947 007/108] cxgb4: Print adapter VPD Part Number instead of Engineering Change field
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

When we attach to adapter, print VPD Part Number instead of Engineering Change field.
Based on original work by Casey Leedom <leedom@chelsio.com>

Signed-off-by: Kumar Sanghvi <kumaras@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit a94cd70521151d107bb069f0c15d1bb7acf302b5)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4.h b/drivers/net/ethernet/chelsio/cxgb4/cxgb4.h
index 0c4edd1..028b5e5 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4.h
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4.h
@@ -66,6 +66,7 @@ enum {
  SERNUM_LEN = 24,    /* Serial # length */
  EC_LEN     = 16,    /* E/C length */
  ID_LEN     = 16,    /* ID length */
+ PN_LEN     = 16,    /* Part Number length */
 };
 
 enum {
@@ -254,6 +255,7 @@ struct vpd_params {
  u8 ec[EC_LEN + 1];
  u8 sn[SERNUM_LEN + 1];
  u8 id[ID_LEN + 1];
+ u8 pn[PN_LEN + 1];
 };
 
 struct pci_params {
diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 85cf3c6..7b626ec 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -5841,8 +5841,8 @@ static void print_port_info(const struct net_device *dev)
       is_offload(adap) ? "R" : "", adap->params.pci.width, spd,
       (adap->flags & USING_MSIX) ? " MSI-X" :
       (adap->flags & USING_MSI) ? " MSI" : "");
- netdev_info(dev, "S/N: %s, E/C: %s\n",
-      adap->params.vpd.sn, adap->params.vpd.ec);
+ netdev_info(dev, "S/N: %s, P/N: %s\n",
+      adap->params.vpd.sn, adap->params.vpd.pn);
 }
 
 static void enable_pcie_relaxed_ordering(struct pci_dev *dev)
diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index 6694df8..13d7690 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -574,7 +574,7 @@ int get_vpd_params(struct adapter *adapter, struct vpd_params *p)
 {
  u32 cclk_param, cclk_val;
  int i, ret, addr;
- int ec, sn;
+ int ec, sn, pn;
  u8 *vpd, csum;
  unsigned int vpdr_len, kw_offset, id_len;
 
@@ -639,6 +639,7 @@ int get_vpd_params(struct adapter *adapter, struct vpd_params *p)
 
  FIND_VPD_KW(ec, "EC");
  FIND_VPD_KW(sn, "SN");
+ FIND_VPD_KW(pn, "PN");
 #undef FIND_VPD_KW
 
  memcpy(p->id, vpd + PCI_VPD_LRDT_TAG_SIZE, id_len);
@@ -648,6 +649,8 @@ int get_vpd_params(struct adapter *adapter, struct vpd_params *p)
  i = pci_vpd_info_field_size(vpd + sn - PCI_VPD_INFO_FLD_HDR_SIZE);
  memcpy(p->sn, vpd + sn, min(i, SERNUM_LEN));
  strim(p->sn);
+ memcpy(p->pn, vpd + pn, min(i, PN_LEN));
+ strim(p->pn);
 
  /*
   * Ask firmware for the Core Clock since it knows how to translate the
-- 
1.7.1